src/Linters/ShoutCaseEnumMembersLinter.hack

```diff
@@ -0,0 +1,85 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST;
+
+use namespace HH\Lib\{C, Keyset, Regex, Str};
+
+final class ShoutCaseEnumMembersLinter extends AutoFixingASTLinter {
+  const type TContext = EnumDeclaration;
+  const type TNode = Enumerator;
+
+  <<__Override>>
+  public function getLintErrorForNode(
+    self::TContext $enum,
+    self::TNode $member,
+  ): ?ASTLintError {
+    $name = $member->getName()->getLastTokenx()->getText();
+    if (self::isShoutCase($name)) {
+      return null;
+    }
+    $fixer = self::isFixableWithoutCollisions($name, $enum)
+      ? () ==> $this->getFixedNode($member)
+      : null;
+    return new ASTLintError(
+      $this,
+      Str\format('Member {%s} is not in SHOUT_CASE', $name),
+      $member,
+      $fixer,
+    );
+  }
+
+  private static function isShoutCase(string $member_name): bool {
+    return Regex\matches($member_name, re"/^[A-Z0-9_]+$/");
+  }
+
+  private static function isFixableWithoutCollisions(
+    string $name,
+    EnumDeclaration $enum,
+  ): bool {
+    $members = $enum->getChildren()['enumerators'] as NodeList<_>;
+    $upper_names = Keyset\map(
+      $members->toVec(),
+      $m ==> $m as Enumerator->getName()->getLastTokenx()->getText()
+        |> Str\uppercase($$),
+    );
+    unset($upper_names[Str\uppercase($name)]);
+    $shout_name = self::transformToShoutCase($name);
+
+    return !C\contains($upper_names, $shout_name);
+  }
+
+  private static function transformToShoutCase(string $name): string {
```

Comment:
Can you extract the logic and reuse it between this linter and `CamelCasedMethodsUnderscoredFunctionsLinter`? I'm not sure which one is better, I think your regex probably handles camel-cased names with acronyms better (GetHTMLVersion) but the other one might handle some other cases better maybe...

Reply:
I tried merging some of the logic, but I could not do so while keeping the same sensible behavior. A function like `fromJSON` should become `from_json`. My regexes make it `from_jSON` (and I uppercase that). The logic requires to make it do something sensible with the casing is completely wasted in the ShoutCase linter, since I effectively don't case about case.

Reply:
I thought the function-name-linter doesn't care about case either. So they could have the same logic, the only difference being that this one calls `Str\uppercas`e at then end, while the other calls `Str\lowercase` at the end.

Reply:
In that case, it is an easy fold.